remme/tp/node_account.py

```
@@ -0,0 +1,107 @@
+            },
+            NodeAccountMethod.CLOSE_MASTERNODE: {
+                PROCESSOR: self._close_masternode,
+                PB_CLASS: EmptyPayload,
+                VALIDATOR: EmptyPayloadForm,
+            },
+            NodeAccountMethod.TRANSFER_FROM_UNFROZEN_TO_OPERATIONAL: {
+                PB_CLASS: NodeAccountInternalTransferPayload,
+                PROCESSOR: self._transfer_from_unfrozen_to_operational,
+                VALIDATOR: NodeAccountInternalTransferPayloadForm,
+            }
+        }
+
+    def _initialize_masternode(self, context, node_account_public_key, internal_transfer_payload):
+        node_account_address = self.make_address_from_data(node_account_public_key)
+
+        node_account = get_node_account_by_address(context, node_account_address)
+
+        if node_account.node_state != NodeAccount.NodeState.Value('NEW'):
+            raise InvalidTransaction('Masternode is already opened or closed.')
+        minimum_stake = int(_get_setting_value(context, 'remme.settings.minimum_stake'))
+
+        if internal_transfer_payload.value < minimum_stake:
+            raise InvalidTransaction('Initial stake is too low.')
+
+        node_account.node_state = NodeAccount.NodeState.Value('OPENED')
+
+        node_account.balance -= internal_transfer_payload.value
+        frozen_part = minimum_stake
+        unfrozen_part = internal_transfer_payload.value - minimum_stake
+        node_account.reputation.frozen += frozen_part
+        node_account.reputation.unfrozen += unfrozen_part
+
+        return {
+            node_account_address: node_account,
+        }
+
+    def _close_masternode(self, context, node_account_public_key, payload):
+        node_account_address = self.make_address_from_data(node_account_public_key)
+
+        node_account = get_node_account_by_address(context, node_account_address)
+
+        if node_account.node_state != NodeAccount.NodeState.Value('OPENED'):
```

Why not `node_account.node_state == closed`? A bit harder to read `!=`, `==` is more readable.

---

Because you can close a masternode only when its state is OPENED. If its state is NEW or CLOSED than method should throw an exception. If I use == it will require two if(s) instead of one or  if-else structure.